### ISSUE TITLE
Vanilla Achievement Support

### ISSUE THREAD

**Issue by GrantGryczan** [bluemagic123/tModLoader#2845]
### Do you intend to personally contribute/program this feature?

No

### I would like to see this change made to improve my experience with

Gameplay as a Player

### Description

Allowing the unlocking of vanilla achievements would be great, even if they don't show up on Steam. I see that there is an issue for achievement support being tracked under #1491, but that includes support for custom mod achievements as well, which is much more far-future than I'd expect vanilla achievement support to be, so I do think vanilla achievement support should be tracked separately.

~~Linking them to the vanilla Terraria Steam achievements (so that unlocking an achievement in tModLoader unlocks it in Terraria, and vice versa) would be exceptionally convenient, although it isn't necessary.~~ (Striked through in consideration of [this issue's first comment](https://github.com/tModLoader/tModLoader/issues/2845#issuecomment-1219882321).)

If linking is not implemented though, then I would at least hope for an option to migrate your vanilla achievement data into tModLoader, like the existing options to migrate players and worlds. I already have many achievements in Terraria and I wouldn't want to have to tediously get them all again in tModLoader.

### What does this proposal attempt to solve or improve?

My own motivation is that it's my first Terraria playthrough, and I'd really like to experience the game with some QOL mods (e.g. Recipe Browser, Magic Storage), but I also want to use achievements as a means to document and check my progress. Also, I find the achievement square next to the inventory very helpful as a constant tip/reminder for what I need to do next to progress. Currently, having both in Terraria 1.4 is impossible. It's a very good vanilla Terraria feature, and I wouldn't expect it to be missing just because I want to play with mods.

**Comment by Chicken-Bones** [comment]
In order to preserve the integrity of vanilla achievements, we can't bring tModLoader achievements to Vanilla. We could (and should) do the opposite.

The main reason nothing has been done (aside from no-one deciding it's the top priority) is that we would like to take a per-world or per-player approach to tML achievements. So you can track your progress with each playthrough. 

We could still keep the overall list, and that could be updated from vanilla, and that would be a suitable 'less difficult' feature that could be implemented separately. I'll leave this open as useful context, and in-case people want to up-vote it, or take it on.

**Comment by zLudociel** [comment]
i'd really like world specific or player specific achievements too
i want to do all achievements in one world but i want to use a few mods like light and shadow (shaders mod)
so this would be awesome

**Comment by GeneralWaffle** [comment]
Yes, I fully agree! Vanilla achievements should be implemented to TML. It would add a lot to any playthrough!

**Comment by Chicken-Bones** [comment]
Needs to be broken down into smaller achievable tasks. Relevant discussion takes place at https://discord.com/channels/103110554649894912/960710141739675728
